src/replication.c

```diff
@@ -1808,27 +1813,24 @@ void readSyncBulkPayload(connection *conn) {
 
         /* RDB loading succeeded if we reach this point. */
         if (server.repl_diskless_load == REPL_DISKLESS_LOAD_SWAPDB) {
-            /* Replica starts to apply data from new master, we must discard the cached
-            * master structure. */
-            serverAssert(server.master == NULL);
-            replicationDiscardCachedMaster();
-
-            /* We want our slaves to resync with us as well, if we have any sub-slaves.
-            * The master already transferred us an entirely different data set and we
-            * have no way to incrementally feed our slaves after that. */
-            disconnectSlaves(); /* Force our slaves to resync with us as well. */
-            freeReplicationBacklog(); /* Don't allow our chained slaves to PSYNC. */
+            if (asyncLoading == 0) {
```

Comment:
i don't think that's right (to put that `if`).
you mean that if `asyncLoading` is true, it means we're still connected to the same master, and thus there's no need to discard the replication backlog and disconnect replicas?
i don't think that's right..
if we are forced to do a full sync (even if that's the same master), we have no way to know what changed, and we can't afford future partial syncs.

anything i'm missing?

Reply:
Oh I see! Reverted, thank you